the-blue-alliance-ios/View Controllers/Events/EventRankingsTableViewController.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  EventRankingsTableViewController.swift
+//  the-blue-alliance-ios
+//
+//  Created by Zach Orr on 5/14/17.
+//  Copyright © 2017 The Blue Alliance. All rights reserved.
+//
+
+import Foundation
+import TBAKit
+import UIKit
+import CoreData
+
+class EventRankingsTableViewController: TBATableViewController {
+
+    override var persistentContainer: NSPersistentContainer! {
+        didSet {
+            updateDataSource()
+        }
+    }
+    var event: Event!
+    var rankingSelected: ((EventRanking) -> ())?
+    
+    // MARK: - View Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: String(describing: EventRankingTableViewCell.self), bundle: nil), forCellReuseIdentifier: EventRankingTableViewCell.reuseIdentifier)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        clearsSelectionOnViewWillAppear = splitViewController!.isCollapsed
+    }
+    
+    // MARK: - Refreshing
+    
+    override func refresh() {
+        removeNoDataView()
+        
+        // First things first... refresh all teams for the district, *then* fetch their rankings
```

Comment:
Thinking about this one... I know we fetch all the teams when fetching district rankings, but I'm wondering if it makes sense here... We might be better off fetching all the teams when we push to the Event VC, and then blocking refreshes for subsequent view controllers (teams, rankings, matches, awards, etc) until we've fetched the teams... for now, let's remove the outer fetch of events and go with the pattern we use in the Awards VC, where when inserting awards we just check if a team exists and if not we make a placeholder team object using the key https://github.com/the-blue-alliance/the-blue-alliance-ios/blob/swift/the-blue-alliance-ios/Core%20Data/Award.swift#L37